kyuubi-server/src/main/scala/org/apache/kyuubi/engine/ProcBuilder.scala

```diff
@@ -319,7 +320,7 @@ trait ProcBuilder {
   protected def getEngineHome(shortName: String): String = {
     val homeKey = s"${shortName.toUpperCase}_HOME"
     // 1. get from env, e.g. SPARK_HOME, FLINK_HOME
-    env.get(homeKey)
+    env.get(homeKey).filter(StringUtils.isNotBlank)
```

Comment:
there is a typo in L310, mind fixing it as well? "directly" => "directory"

Reply:
fixed